### ISSUE TITLE
"Open" file dialog should be easier to use with a keyboard

### ISSUE THREAD

**Issue by salty-horse** [curiousdannii/asyncglk#47]
Since parser games primarily use the keyboard, one should be able to use the keyboard when saving and restoring.

With saving, this works fine for the default use-case. When typing the command "save", a dialog opens with the focus on the file name field. I can type something and press the enter key.

With the "open" dialog, it feels impossible to navigate and use. When I type "restore", I can a dialog with no focus visible anywhere. Here are some annoyances I have from this point:
1. Pressing enter closes it with an error.
2. Pressing tab cycles the focus between the various elements, and each file in the list. But having a focus in the file isn't enough. I have to press the space bar to actually select it.
3. When the focus is on the file list, the up and down arrow keys don't work - they should.

I think when the "open" dialog first appears, the first file in the list should already be selected. I should just be able to press the up and down arrow keys to select a different file (or directory) and then press the enter key to select it.

**Comment by curiousdannii** [comment]
Absolutely! Thank you for this feedback, I mustn't have tried using it much without mouse/touch before.

**Comment by curiousdannii** [comment]
I think I've largely fixed this. The dialog focuses now seem appropriate, the arrow keys work etc. But let me know if you have any more feedback on it.

**Comment by salty-horse** [comment]
Much better! Thank you!

A few minor inconsistencies:
1. The focus order in the top area doesn't match the left-to-right order. If I "shift-tab" from the top file in the list, the focus goes to the directory name in the path breadcrumbs, when I expect it to go to the "new folder" button.
2. When opening a directory (either in the file list or the path breadcrumbs at the top), the resulting file list should have focus - it doesn't.

**Comment by salty-horse** [comment]
Another issue:
If there are a lot of files in the directory, and then you pick one to have the blue highlight, and start pressing the down button, the file that has the blue highlight will slowly disappear as the scroll moves down faster than it.

The blue-highlighted item should always be fully visible.

https://github.com/user-attachments/assets/74928cc7-4528-4bc1-b9ef-315e055c4828

It would also be nice to have a way to sort the save files. You usually want to sort them by modification date, rather than alphabetically, since newer save files are more frequently used.

**Comment by salty-horse** [comment]
The save dialog opens with a focus on the filename field. I think that pressing up/down from that position should start selecting files in the list above the field, so you can easily replace an existing save file.

(I think the default behavior of operating systems is for up/down is to open a drop-down of historical entries, but that's less useful)
